runtime-interface/src/lib.rs

```diff
@@ -171,42 +171,37 @@ pub trait GearRI {
     /// If `protect` argument is true then restrict all accesses to page,
     /// else allows read and write accesses.
     fn mprotect_lazy_pages(wasm_mem_addr: u64, protect: bool) -> Result<(), MprotectError> {
-        let mut prev_page = 0u32;
-        let mut size_in_pages = 0u32;
-        let mut lazy_pages = gear_lazy_pages::get_lazy_pages_numbers();
+        let lazy_pages = gear_lazy_pages::get_lazy_pages_numbers();
+        if lazy_pages.is_empty() {
+            return Ok(());
+        }
 
         if let Some(page) = lazy_pages.last() {
             // This case is impossible in real live, so just returns err and does nothing.
-            if *page == u32::MAX || *page == u32::MAX - 1 {
+            if *page >= u32::MAX - 1 {
                 return Err(MprotectError::PageError);
             }
         }
 
-        // We add this page num to lazy pages in order to be able correctly
-        // finish lazy pages handling in loop. This last page won't be
-        // handled.
-        lazy_pages.push(u32::MAX);
+        let mprotect = |start, count, protect: bool| unsafe {
+            let addr = wasm_mem_addr + (start * PageNumber::size()) as u64;
+            let size = count * PageNumber::size();
+            sys_mprotect_interval(addr, size, !protect, !protect, false)
```

Comment:
I've just rewritten the function. @grishasobol please check whether there is no error in `protect` vs `!protect`.

Reply:
we definitely need a test(s) for comfortable refactoring =)

Reply:
I don't know how to efficiently test mprotect. Need assistance from @grishasobol